plan/point_select_plan.go

```diff
@@ -0,0 +1,435 @@
+			continue
+		}
+		schema := buildSchemaFromFields(ctx, tbl, selStmt.Fields.Fields)
+		if schema == nil {
+			return nil
+		}
+		p := newPointSelectPlan(ctx, schema, tbl)
+		p.IndexInfo = idxInfo
+		p.IndexValues = idxValues
+		return p
+	}
+	return nil
+}
+
+func newPointSelectPlan(ctx sessionctx.Context, schema *expression.Schema, tbl *model.TableInfo) *PointSelectPlan {
+	p := &PointSelectPlan{
+		basePlan: newBasePlan(ctx, "Point_Select"),
+		schema:   schema,
+		TblInfo:  tbl,
+	}
+	p.stats = &statsInfo{count: 1}
+	return p
+}
+
+func checkFastPlanPrivilege(ctx sessionctx.Context, fastPlan *PointSelectPlan, checkTypes ...mysql.PrivilegeType) error {
+	pm := privilege.GetPrivilegeManager(ctx)
+	if pm == nil {
+		return nil
+	}
+	dbName := ctx.GetSessionVars().CurrentDB
+	for _, checkType := range checkTypes {
+		if !pm.RequestVerification(dbName, fastPlan.TblInfo.Name.L, "", checkType) {
+			return errors.New("privilege check fail")
+		}
+	}
+	return nil
+}
+
+func buildSchemaFromFields(ctx sessionctx.Context, tbl *model.TableInfo, fields []*ast.SelectField) *expression.Schema {
+	dbName := model.NewCIStr(ctx.GetSessionVars().CurrentDB)
+	columns := make([]*expression.Column, 0, len(tbl.Columns)+1)
+	if len(fields) == 1 && fields[0].WildCard != nil {
+		for _, col := range tbl.Columns {
+			columns = append(columns, colInfoToColumn(dbName, tbl.Name, col, len(columns)))
+		}
+		return expression.NewSchema(columns...)
+	}
+	if len(fields) > 0 {
+		for _, field := range fields {
+			colNameExpr, ok := field.Expr.(*ast.ColumnNameExpr)
+			if !ok {
+				return nil
+			}
+			col := findCol(tbl, colNameExpr.Name)
+			if col == nil {
+				return nil
+			}
+			columns = append(columns, colInfoToColumn(dbName, tbl.Name, col, len(columns)))
+		}
+		return expression.NewSchema(columns...)
+	}
+	// fields len is 0 for update and delete.
+	var handleCol *expression.Column
+	for _, col := range tbl.Columns {
+		column := colInfoToColumn(dbName, tbl.Name, col, len(columns))
+		if tbl.PKIsHandle && mysql.HasPriKeyFlag(col.Flag) {
+			handleCol = column
+		}
+		columns = append(columns, column)
+	}
+	if handleCol == nil {
+		handleCol = colInfoToColumn(dbName, tbl.Name, model.NewExtraHandleColInfo(), len(columns))
+		columns = append(columns, handleCol)
+	}
+	schema := expression.NewSchema(columns...)
+	schema.TblID2Handle = make(map[int64][]*expression.Column)
+	schema.TblID2Handle[tbl.ID] = []*expression.Column{handleCol}
+	return schema
+}
+
+func getSingleTableName(tableRefs *ast.TableRefsClause) *ast.TableName {
+	if tableRefs == nil || tableRefs.TableRefs == nil || tableRefs.TableRefs.Right != nil {
+		return nil
+	}
+	tblSrc, ok := tableRefs.TableRefs.Left.(*ast.TableSource)
+	if !ok {
+		return nil
+	}
+	if tblSrc.AsName.L != "" {
+		return nil
+	}
+	tblName, ok := tblSrc.Source.(*ast.TableName)
+	if !ok {
+		return nil
+	}
+	return tblName
+}
+
+func getNameValuePairs(nvPairs []nameValuePair, expr ast.ExprNode) []nameValuePair {
```

Comment:
add more comments for this function, for example: what kind of expr functions this functions handles?